### PR TITLE
refactor: convert security pallet to FRAME v2

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

**Comment by sander2** [comment]
Probably it makes sense to make a pr for converting all pallets at once, since it would be good to keep everything consistent. Or do you think that is too much work right now?

**Comment by gregdhill** [comment]
> Probably it makes sense to make a pr for converting all pallets at once, since it would be good to keep everything consistent. Or do you think that is too much work right now?

Since the change is mostly superficial I would prefer to do one pallet at a time to make reviews easier. I also wanted to minimize the potential for merge conflicts with ongoing work since the changes are quite invasive.

File: crates/security/src/lib.rs [hunk 1]
```diff
@@ -37,42 +36,55 @@ use sha2::{Digest, Sha256};
...
     }
-}
 
-// The pallet's dispatchable functions.
-decl_module! {
-    pub struct Module<T: Config> for enum Call where origin: T::Origin {
-        type Error = Error<T>;
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    #[pallet::metadata(T::AccountId = "StatusCode")]
```
Comment (sander2) [inline crates/security/src/lib.rs:57]: What does this line do?

**Review by sander2 — APPROVED** [review]

File: crates/security/src/lib.rs (hunk 1 above)
Comment (gregdhill) [reply crates/security/src/lib.rs]: This was an oversight, but for reference > https://crates.parity.io/frame_support/attr.pallet.html#event-palletevent-optional

**Pull request merged**